projects/saturn/src/main/java/io/fairspace/saturn/services/views/ViewStoreClientFactory.java

```diff
@@ -103,62 +102,103 @@ void ensureTableExists(Table table) throws SQLException {
+                .collect(Collectors.toList());
+        try {
+            log.debug("New columns: {}", new ObjectMapper().writeValueAsString(newColumns));
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
+        }
+        // Update existing table
+        if (!newColumns.isEmpty()) {
+            connection.setAutoCommit(true);
+            var command = newColumns.stream()
+                    .map(column -> String.format("alter table %s add column %s %s",
+                            table.name, column.name, databaseTypeForColumnType(column.type)))
+                    .collect(Collectors.joining("; "));
+            log.debug(command);
+            connection.createStatement().execute(command);
+            connection.setAutoCommit(false);
+            log.info("Table {} updated.", table.name);
+        }
+    }
+
+    private void createTable(Table table, Connection connection) throws SQLException {
+        // Create new table
+        connection.setAutoCommit(true);
+        var columnSpecification = table.columns.stream()
+                .filter(column -> !column.type.isSet())
+                .map(column -> String.format("%s %s", column.name, databaseTypeForColumnType(column.type)))
+                .collect(Collectors.joining(", "));
+        var keys = table.columns.stream()
+                .filter(column -> column.type == ColumnType.Identifier)
+                .map(column -> column.name)
+                .collect(Collectors.joining(", "));
+        var command = String.format("create table %s ( %s, primary key ( %s ) )", table.name, columnSpecification,
+                keys);
+        log.debug(command);
+        connection.createStatement().execute(command);
+        connection.setAutoCommit(false);
+        log.info("Table {} created.", table.name);
+    }
+
+    private void createIndexes(Table table, Connection connection) throws SQLException {
+        
+        var keys = table.columns.stream()
+                .filter(column -> column.type == ColumnType.Identifier)
+                .map(column -> column.name)
+                .toList();
+
+        connection.setAutoCommit(true);
+
+        for (var column : keys) {
+            var indexName = String.format("%1$s_%2$s_idx", table.name, column);
+            var command = String.format("CREATE INDEX IF NOT EXISTS %3$s ON public.%1$s (%2$s)", table.name, column, indexName);
+
+            log.debug(command);
+            connection.createStatement().execute(command);
+            log.info("Index {} created.", indexName);
+        }
+        
+        connection.setAutoCommit(false);
+    }
+
     void validateViewConfig(ViewsConfig.View view) {
-        if (view.columns.stream().anyMatch(column ->
-                column.name.equalsIgnoreCase("id"))) {
+        if (view.columns.stream().anyMatch(column -> column.name.equalsIgnoreCase("id"))) {
```

Comment:
please, extract variable with proper naming for batter readability

Reply:
and in other if conditions below

Reply:
for complex cases I would agree, however in this case it is just a single line with a single string comparison, I think any developer will understand and will not improve readability since you would add 4 lines of code just for 4 string comparisons.